src/compiler/parser-quirk-vafirst.bas

```diff
@@ -271,22 +274,61 @@ function cVALISTFunct _
 		function = astNewMACRO( AST_OP_VA_ARG, expr, NULL, dtype, subtype )
 
 	else
-		'' pointer expression: modify expr variable & return value of expr variable before modification
+
+		'' pointer expression: modify expr/variable & return value of expr variable before modification
 		'' based on:
 		'' #define __va_argsiz(t) (((sizeof(t) + sizeof(int) - 1) / sizeof(int)) * sizeof(int))
 		'' #define va_arg(ap, t) (((ap) = (ap) + __va_argsiz(t)), *((t*) (void*) ((ap) - __va_argsiz(t))))
+		'' assign to a temp variable if anything but a variable
+
+		dim tree as ASTNODE ptr = NULL
+		dim temp as FBSYMBOL ptr = any
+
+		select case astGetClass( expr )
+		case AST_NODECLASS_VAR, AST_NODECLASS_IDX, AST_NODECLASS_FIELD
+			'' use expr as-is, no side effects expected...
+			 
+		case AST_NODECLASS_DEREF
+			'' copy the reference to a variable, to prevent side effects
```

Comment:
maybe `astRemSideFx()` can be re-used to build the assignment/temp var

Reply:
Yeah, I saw `astRemSideFx()`, `astMakeRef`, `hGetExprRef`, and this new code for `cva_arg()` all kind doing similar things.  I'll have another look.